app/models/api_model.php

```diff
@@ -61,7 +65,26 @@
                     $short_desc = $set_limit(80, $short_desc($translate));
 
                     if ($desc && $title && $short_desc) {
-                        $done[] = $lang;
+                        /* To be included in whatsnew complete locales, a locale
+                         * must have translated the other strings first.
+                         * If other strings are complete but whatsnew is uncomplete,
+                         * consider the locale done.
+                         */
+                        if (isset($request->query['section']) && $request->query['section'] == 'whatsnew') {
+                            if ($request->query['channel'] == 'next' || $request->query['channel'] == 'release') {
+                                $translate_whatsnew = new Translate($lang, $project->getLangFile($request->query['store'], $request->query['channel'], 'whatsnew'));
+
+                                // Check if whatsnew is done
+                                if ($translate_whatsnew->isFileTranslated() && $set_limit(500, $whatsnew($translate))) {
+                                    $done[] = $lang;
+                                }
+                            } else {
+                                http_response_code(400);
+                                $json = ['error' => 'No \'What\'s new\' section for this channel.'];
```

Comment:
For Google, we should probably always have a whatsnew field in the `api/google/translation/beta/fi/` for consistency with release and because we may translate the what's new section there in the future. Let's just have an empty field for now. I prefer that the schema we have for Google json APIs doesn't change between channels and that we don't complexify the code to make them different.